src/main/Config/AppConfig.java

```diff
@@ -31,11 +35,23 @@ public static Javalin appFactory(DeploymentLevel deploymentLevel) {
     LogFactory l = new LogFactory();
     l.createLogger();
 
+    /* Create the Encryption Controller for encryption */
+    EncryptionController encryptionController;
+    try {
+      encryptionController = new EncryptionController(db);
+    } catch (GeneralSecurityException | IOException e) {
+      // Is this correct? #Connor
+      System.err.println(e.getStackTrace());
+      System.exit(0);
+      return null;
+    }
+
     // We need to instantiate the controllers with the database.
-    OrganizationController orgController = new OrganizationController(db);
-    UserController userController = new UserController(db);
-    AccountSecurityController accountSecurityController = new AccountSecurityController(db);
-    PdfController pdfController = new PdfController(db);
+    OrganizationController orgController = new OrganizationController(db, encryptionController);
```

Comment:
Why are we passing the instance of EncryptionController to the constructor of every other controller? This is inconsistent with the way we have been creating controllers. It seems like most of the methods in EncryptionController can be used statically, since they do not rely on any internal state. See my comments on EncryptionController.java.

I actually don't think that EncryptionController is a controller at all. It does not have handlers. Can we rename this class EncryptionUtils that has a bunch of static methods? We don't need to pass it to the constructors of the other controllers. Just invoke the static methods.

Reply:
Actually, we need to call generateAEAD, which has local state, so I am using a singleton design pattern

Reply:
Let me know if it is good

Reply:
If you have local state, then initialize it in AppConfig, but you should not initialize it in the constructor of EncryptionController. It breaks the design pattern of no code in constructors

Reply:
How will it be referenced inside of Encryption Controller? There only needs to be one AEAD used throughout the whole program